posts/views.py

```diff
@@ -115,3 +115,11 @@ def delete_tag(request, id):
         post.tag.remove(removable_tag)
         post.save()
     return Response({}, status=status.HTTP_200_OK)
+
+
+@api_view(['GET'])
+@permission_classes([p.AllowAny, ])
+def search_by_tag(request):
+    searched_tag = Tag.objects.filter(tag=request.data['tag']).first()
```

Comment:
First, you wrote a search for posts by tags. It's OK, you can leave it.  But you need to search by tag name so that you can get a list of existing tags by pattern. 
Second, use ILIKE to make the search work not by the whole word, but by part

Reply:
Done.